### DISCUSSION TITLE
Support Duration

### DISCUSSION THREAD

**Discussion by fischerscode** [serverpod/serverpod#310] [Ideas]
Maybe it's just me, but I think it would be nice to support `Duration` just like `DateTime` is.

**What do you think?**

**Comment by aldychris** [comment]
Im a bit not clear, the use case of the Duration here.. is it for when we want to filter the data?  Something like  
`
var companies = await Company.find(
  tCompany,
  where: (t) => t.createdDate.isBefore(duration),
);
`

**Reply by fischerscode** [reply]
I had storing them in the database as well as sending them to the client in mind. Database expressions might be something like "equals" (same duration), "isLonger", "isShorter",  ...

However, supporting more datetime native dart methods as database expressions (like your example) is something I would love to be added as well.

**Comment by vlidholt** [comment]
Great suggestion, will add Duration support in the future, cannot make any promises for the 1.0 release as its almost around the corner now.

**Reply by fischerscode** [reply]
So a PR might be accepted? Just in case I find time...

**Reply by vlidholt** [reply]
Yes :)
